L1Trigger/TrackFindingTMTT/interface/GlobalCacheTMTT.h

```diff
@@ -0,0 +1,56 @@
+#ifndef L1Trigger_TrackFindingTMTT_GlobalCacheTMTT_h
+#define L1Trigger_TrackFindingTMTT_GlobalCacheTMTT_h
+
+// Data shared across all streams by TMTT L1 tracking.
+//
+// Provides the python configuration parameters
+// & optional histogramming/debugging.
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "L1Trigger/TrackFindingTMTT/interface/Settings.h"
+#include "L1Trigger/TrackFindingTMTT/interface/HTrphi.h"
+#include "L1Trigger/TrackFindingTMTT/interface/StubWindowSuggest.h"
+#include "L1Trigger/TrackFindingTMTT/interface/TrackerModule.h"
+#include "L1Trigger/TrackFindingTMTT/interface/Histos.h"
+
+#include <list>
+#include <memory>
+
+namespace tmtt {
+
+  class GlobalCacheTMTT {
+  public:
+    GlobalCacheTMTT(const edm::ParameterSet& iConfig)
+        : settings_(iConfig),              // Python configuration params
+          htRphiErrMon_({0., 0, 0, 0}),    // rphi HT error monitoring
+          stubWindowSuggest_(&settings_),  // Recommend FE stub window sizes.
+          hists_(&settings_)               // Histograms
+    {
+      hists_.book();
+    }
+
+    // Get functions
+    Settings& settings() const { return settings_; }
+    HTrphi::ErrorMonitor& htRphiErrMon() const { return htRphiErrMon_; }
```

Comment:
It would be a bit clearer to declare all the `std::atomic<T>` in `ErrorMonitor` as `mutable atomic<T>` and pass around `const HTrphi::ErrorMonitor&`.

Reply:
Could you please clarify why? With htRphiErrMon_ declared mutable inside GlobalCacheTMTT, it is immediately obvious  to anyone reading GlobalCacheTMTT that all the data members inside htRphiErrMon_ are mutable. But if we don't declare htRphiErrMon_ mutable, and instead declare the data members inside it mutable, someone reading GlobalCacheTMTT would not immediately appreciate this. They would have to read the HTrphi class to do so, taking time.

Reply:
Right, on a second thought I came to the same conclusion that it is indeed better pass `HTrphi::ErrorMonitor` as non-`const` to show the intention that it is modified.

Reply:
left unchanged